bessctl/conf/testing/run_module_tests.bess

```diff
@@ -0,0 +1,166 @@
+
+# These are just for the crash test
+crash_test_packets = [
+    gen_packet(scapy.UDP, '172.16.100.1', '10.0.0.1'),
+    gen_packet(scapy.UDP, '172.12.55.99', '12.34.56.78'),
+    gen_packet(scapy.UDP, '172.12.55.99', '10.0.0.1'),
+    gen_packet(scapy.UDP, '172.16.100.1', '12.34.56.78'),
+    gen_packet(scapy.TCP, '172.12.55.99', '12.34.56.78'),
+    gen_packet(scapy.UDP, '192.168.1.123', '12.34.56.78')
+]
+
+## TEST FUNCTIONS ##
+
+# All this does is subject a module to load and make sure it
+# doesn't crash
+
+
+def crash_test(module, num_input_ports, num_output_ports):
+    for i in range(num_input_ports):
+        src = Source()
+        src -> Rewrite(templates=crash_test_packets) -> i:module
+    for i in range(num_output_ports):
+        sink = Sink()
+        module:i -> sink
+
+    time.sleep(5)
+
+
+## RUN TEST LOOP ##
+
+path = os.path.dirname(os.path.realpath(sys.argv[0]))
+for file in glob.glob(path + "/conf/testing/module_tests/*.py"):
+    bess.reset_all()
+    CRASH_TEST_INPUTS = []
+    OUTPUT_TEST_INPUTS = []
+    CUSTOM_TEST_FUNCTIONS = []
+
+    print("-- Running Module Tests for " + file + " --")
+    while True:  # wrap in this scoping to make sure variables don't leak into this script
+        exec(sugar.xform_file(file))
+        break
+
+    crash_testid = 1
+    for test in CRASH_TEST_INPUTS:
+        crash_test(test[0], test[1], test[2])
+        print("   " +
+              str(test[0]) +
+              " crash test " +
+              str(crash_testid) +
+              ": PASS")
+        crash_testid += 1
+
+    output_testid = 1
+    for test in OUTPUT_TEST_INPUTS:
+        input_ports = []
+        output_ports = []
+        sockets = []
+
+        for port_num in range(max(test[1], test[2])):
+            sockname = str(output_testid) + "moduletesting-" + \
+                SCRIPT_STARTTIME + str(port_num)
+            socket_port, mysocket = gen_socket_and_port(sockname)
+            input_ports.append(PortInc(port=sockname))
+            output_ports.append(PortOut(port=sockname))
+            sockets.append(mysocket)
+            if (port_num < test[1]):
+                input_ports[port_num] -> port_num:test[0]
+            if (port_num < test[2]):
+                test[0]:port_num -> output_ports[port_num]
+
+        bess.resume_all()
+
+        for test_case in test[3]:
+            input_port = test_case["input_port"]
+            output_port = test_case["output_port"]
+            input_pkt = test_case["input_packet"]
+            output_pkt = test_case["output_packet"]
+            if(input_pkt is not None):
+                sockets[input_port].send(str(input_pkt))
+            try:
+                return_pkt = scapy.Ether(sockets[output_port].recv(2048))
+            except socket.timeout:
+                return_pkt = None
+            if(not ((return_pkt is None and output_pkt is None) or
+                    (return_pkt == scapy.Ether(output_pkt)))):
+                print("Test failed!\n")
```

Comment:
You can use stderr for failure-related outputs (thus Travis can highlight them properly).